src/components/date-range/date-range.test.js

```diff
@@ -0,0 +1,370 @@
+/* eslint-disable react/prop-types */
+import * as React from "react";
+import DateRange from "./date-range.component";
+
+import { getDataElementByValue, icon } from "../../../cypress/locators";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+
+const testCypressText = "test_cypress";
+
+const DateRangeCustom = ({ onChange, onBlur, ...props }) => {
+  const [state, setState] = React.useState(["01/10/2016", "30/10/2016"]);
+
+  const handleChange = (evt) => {
+    const newValue = [
+      evt.target.value[0].formattedValue,
+      evt.target.value[1].formattedValue,
+    ];
+
+    if (onChange) {
+      onChange(evt);
+    }
+
+    setState(newValue);
+  };
+
+  const handleOnBlur = (evt) => {
+    if (onBlur) {
+      onBlur(evt);
+    }
+  };
+
+  return (
+    <DateRange
+      startLabel="start label"
+      endLabel="end label"
+      onChange={handleChange}
+      onBlur={handleOnBlur}
+      value={state}
+      name={testCypressText}
+      id={testCypressText}
+      {...props}
+    />
+  );
+};
+
+const testData = ["mp150ú¿¡üßä", "!@#$%^*()_+-=~[];:.,?{}&\"'<>"];
+const START_DATE_RANGE_INDEX = 0;
+const END_DATE_RANGE_INDEX = 1;
+const ERROR_COLOUR = "rgb(205, 56, 75)";
+const WARN_COLOUR = "rgb(239, 103, 0)";
+const INFO_COLOUR = "rgb(0, 96, 167)";
+
+const checkInputColor = (index, color) => {
+  getDataElementByValue("input")
+    .eq(index)
```

Comment:
I think here we should start with data-component="date-range" and then go deeper to the input element

Reply:
fixed